src/backend/distributed/utils/node_metadata.c

```diff
@@ -630,7 +630,7 @@ get_shard_id_for_distribution_column(PG_FUNCTION_ARGS)
 		List *shardIntervalList = LoadShardIntervalList(relationId);
 		if (shardIntervalList == NIL)
 		{
-			PG_RETURN_INT64(NULL);
+			PG_RETURN_INT64(0);
```

Comment:
I think this should be `PG_RETURN_NULL()`. I think `0` is a valid shard Id, so it doesn't mean the function didn't find any shards.

`PG_RETURN_NULL()` sets `fcinfo->isnull` to `true` before returning from function, so postgres knows this function returned null. `PG_RETURN_INT64()` doesn't set that.

Reply:
It seems that our shard Ids start from `102008`, but anyway I‌ think returning `NULL` for absence makes more sense than returning 0.

Reply:
Changing this function to return `NULL` rather than `0` would be a change in a public API, no?

Reply:
I'm not sure what the right thing to do here is. @jasonmp85 is technically right, the only kind of right, this has a chance of breaking user code. I think we're doing a v7.3 next and miner versions aren't allowed to change the API?

On the other hand, the chance of breaking user code is pretty small and returning `NULL` is much cleaner.